src/main/java/gift/auth/service/AuthService.java

```diff
@@ -0,0 +1,37 @@
+package gift.auth.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gift.auth.dto.AuthTokenResponseDto;
+import gift.auth.dto.KakaoTokenResponse;
+import gift.member.entity.Member;
+import gift.member.service.MemberService;
+import gift.security.config.JwtProvider;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+
+    private final KakaoOAuthService kakaoOAuthService;
+    private final MemberService memberService;
+    private final JwtProvider jwtProvider;
+    private final ObjectMapper objectMapper;
+
+    public AuthTokenResponseDto loginWithKakao(String code) throws JsonProcessingException {
+        KakaoTokenResponse kakaoToken = kakaoOAuthService.requestAccessToken(code);
+
+        String profileJson = kakaoOAuthService.requestRawUserInfo(kakaoToken.getAccessToken());
+
+        JsonNode root = objectMapper.readTree(profileJson);
+        String email = root.path("kakao_account").path("email").asText();
+        String nickname = root.path("properties").path("nickname").asText();
```

Comment:
```suggestion
        KakaoUserResponse response = kakaoOAuthService.requestRawUserInfo(kakaoToken.getAccessToken());
```

이렇게 리팩토링 해보는건 어떨까요?
kakaoOAuthService 에서 주는 api 스펙에서 우리가 필요한 데이터만 넘겨받는것도 좋을것 같아요 🤔

Reply:
네네 혼자 잘못 이해하는 바람에 반영이 늦었습니다 ㅜ.ㅜ..!!

`AuthService`에서 파싱 로직을 신경 쓰지 않고, DTO만 받아 처리할 수 있게 리팩토링 완료하였습니다~!

다시 한번 늦은 시간에 잔뜩 질문 드렸는데 답변 감사드려요 ☺️

Reply:
> 네네 혼자 잘못 이해하는 바람에 반영이 늦었습니다 ㅜ.ㅜ..!!

관련해서 개인적으로 궁금한 것이 있는데, 혼자 작업을 진행하다가 막혔을 때 보통 얼마 정도 스스로 고민해 보신 후에 도움을 요청하시나요? 또, 도움을 요청하는 나름의 기준(고민 시간, 시도 횟수 등..)이 있다면 공유해주시면 감사하겠습니다..!